### ISSUE TITLE
`useless-comparison (B015)` + `pytest` and expecting an exception

### ISSUE THREAD

**Issue by stdedos** [astral-sh/ruff#23733]
### Question

I understand the usual flow of "every test should be an `assert`", but the documentation page does not address this.

Based on pytest's documentation https://docs.pytest.org/en/stable/reference/reference.html#pytest-raises, the example the offer is this:

```py
import pytest
with pytest.raises(ZeroDivisionError):
   1/0
```

which also triggers `useless-comparison (B015)` in the https://play.ruff.rs/ playground.

At the very least, adding such an example to the documentation would help future readers.

I already looked for it https://docs.astral.sh/ruff/rules/useless-comparison, https://github.com/astral-sh/ruff/issues/?q=is%3Aissue%20pytest%20useless-comparison but I didn't find anything useful

### Version

v0.15.4 (and v0.14.14 too)

**Comment by ntBre** [comment]
This example doesn't seem to flag B015 in the [playground](https://play.ruff.rs/b7a56565-f1ef-4fe4-8b9a-1e350104b687). However, it does flag [useless-expression (B018)](https://docs.astral.sh/ruff/rules/useless-expression/#useless-expression-b018), which I assume you meant instead. B018 does include an example of intentionally discarding an expression in its [docs](https://docs.astral.sh/ruff/rules/useless-expression/#known-problems), something like this in this case:

```py
import pytest
with pytest.raises(ZeroDivisionError):
   _ = 1/0
```
